File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Options Page/C#/UnitTests/VSCodeGenAccessors.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Microsoft.Samples.VisualStudio.IDE.OptionsPage.UnitTests
{
[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class BaseAccessor {
    
    protected Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject m_privateObject;
    
    protected BaseAccessor(object target, Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType type) {
        m_privateObject = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject(target, type);
    }
    
    protected BaseAccessor(Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType type) : 
            this(null, type) {
    }
    
    internal virtual object Target {
        get {
            return m_privateObject.Target;
        }
    }
    
    public override string ToString() {
        return this.Target.ToString();
    }
    
    public override bool Equals(object obj) {
        if (typeof(BaseAccessor).IsInstanceOfType(obj)) {
            obj = ((BaseAccessor)(obj)).Target;
        }
        return this.Target.Equals(obj);
    }
    
    public override int GetHashCode() {
        return this.Target.GetHashCode();
    }
}


[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPagePackageAccessor : BaseAccessor {
    
    protected static Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType m_privateType = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(typeof(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPagePackageCS));
    
    internal Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPagePackageAccessor(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPagePackageCS target) : 
            base(target, m_privateType) {
    }
    
    internal void Initialize() {
        object[] args = new object[0];
        m_privateObject.Invoke("Initialize", new System.Type[0], args);
    }
}
[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageCustomAccessor : BaseAccessor {
    
    protected static Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType m_privateType = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(typeof(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom));
    
    internal Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageCustomAccessor(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom target) : 
            base(target, m_privateType) {
    }
    
    internal string selectedImagePath {
        get {
            string ret = ((string)(m_privateObject.GetField("selectedImagePath")));
            return ret;
        }
        set {
            m_privateObject.SetField("selectedImagePath", value);
        }
    }
    
    internal global::System.Windows.Forms.IWin32Window Window {
        get {
            global::System.Windows.Forms.IWin32Window ret = ((global::System.Windows.Forms.IWin32Window)(m_privateObject.GetProperty("Window")));
            return ret;
        }
    }
}
[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsCompositeControlAccessor : BaseAccessor {
    
    protected static Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType m_privateType = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(typeof(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsCompositeControl));
    
    internal Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsCompositeControlAccessor(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsCompositeControl target) : 
            base(target, m_privateType) {
    }
    
    internal global::System.Windows.Forms.PictureBox pictureBox {
        get {
            global::System.Windows.Forms.PictureBox ret = ((global::System.Windows.Forms.PictureBox)(m_privateObject.GetField("pictureBox")));
            return ret;
        }
        set {
            m_privateObject.SetField("pictureBox", value);
        }
    }
    
    internal global::System.Windows.Forms.OpenFileDialog openImageFileDialog {
        get {
            global::System.Windows.Forms.OpenFileDialog ret = ((global::System.Windows.Forms.OpenFileDialog)(m_privateObject.GetField("openImageFileDialog")));
            return ret;
        }
        set {
            m_privateObject.SetField("openImageFileDialog", value);
        }
    }
    
    internal global::System.Windows.Forms.Button buttonChooseImage {
        get {
            global::System.Windows.Forms.Button ret = ((global::System.Windows.Forms.Button)(m_privateObject.GetField("buttonChooseImage")));
            return ret;
        }
        set {
            m_privateObject.SetField("buttonChooseImage", value);
        }
    }
    
    internal global::System.Windows.Forms.Button buttonClearImage {
        get {
            global::System.Windows.Forms.Button ret = ((global::System.Windows.Forms.Button)(m_privateObject.GetField("buttonClearImage")));
            return ret;
        }
        set {
            m_privateObject.SetField("buttonClearImage", value);
        }
    }
    
    internal global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom customOptionsPage {
        get {
            global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom ret = ((global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom)(m_privateObject.GetField("customOptionsPage")));
            return ret;
        }
        set {
            m_privateObject.SetField("customOptionsPage", value);
        }
    }
    
    internal global::System.ComponentModel.Container components {
        get {
            global::System.ComponentModel.Container ret = ((global::System.ComponentModel.Container)(m_privateObject.GetField("components")));
            return ret;
        }
        set {
            m_privateObject.SetField("components", value);
        }
    }
    
    internal void Dispose(bool disposing) {
        object[] args = new object[] {
                disposing};
        m_privateObject.Invoke("Dispose", new System.Type[] {
                    typeof(bool)}, args);
    }
    
    internal void InitializeComponent() {
        object[] args = new object[0];
        m_privateObject.Invoke("InitializeComponent", new System.Type[0], args);
    }
    
    internal void OnChooseImage(object sender, global::System.EventArgs e) {
        object[] args = new object[] {
                sender,
                e};
        m_privateObject.Invoke("OnChooseImage", new System.Type[] {
                    typeof(object),
                    typeof(global::System.EventArgs)}, args);
    }
    
    internal void OnClearImage(object sender, global::System.EventArgs e) {
        object[] args = new object[] {
                sender,
                e};
        m_privateObject.Invoke("OnClearImage", new System.Type[] {
                    typeof(object),
                    typeof(global::System.EventArgs)}, args);
    }
    
    internal void RefreshImage() {
        object[] args = new object[0];
        m_privateObject.Invoke("RefreshImage", new System.Type[0], args);
    }
}
[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageGeneralAccessor : BaseAccessor {
    
    protected static Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType m_privateType = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(typeof(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageGeneral));
    
    internal Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageGeneralAccessor(global::Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageGeneral target) : 
            base(target, m_privateType) {
    }
    
    internal int optionCustomInteger {
        get {
            int ret = ((int)(m_privateObject.GetField("optionCustomInteger")));
            return ret;
        }
        set {
            m_privateObject.SetField("optionCustomInteger", value);
        }
    }
    
    internal global::System.Drawing.Size optionCustomSize {
        get {
            global::System.Drawing.Size ret = ((global::System.Drawing.Size)(m_privateObject.GetField("optionCustomSize")));
            return ret;
        }
        set {
            m_privateObject.SetField("optionCustomSize", value);
        }
    }
    
    internal string optionCustomString {
        get {
            string ret = ((string)(m_privateObject.GetField("optionCustomString")));
            return ret;
        }
        set {
            m_privateObject.SetField("optionCustomString", value);
        }
    }
    
    internal void OnActivate(global::System.ComponentModel.CancelEventArgs e) {
        object[] args = new object[] {
                e};
        m_privateObject.Invoke("OnActivate", new System.Type[] {
                    typeof(global::System.ComponentModel.CancelEventArgs)}, args);
    }
    
    internal void OnClosed(global::System.EventArgs e) {
        object[] args = new object[] {
                e};
        m_privateObject.Invoke("OnClosed", new System.Type[] {
                    typeof(global::System.EventArgs)}, args);
    }
    
    internal void OnDeactivate(global::System.ComponentModel.CancelEventArgs e) {
        object[] args = new object[] {
                e};
        m_privateObject.Invoke("OnDeactivate", new System.Type[] {
                    typeof(global::System.ComponentModel.CancelEventArgs)}, args);
    }
    
    internal void OnApply(Microsoft_VisualStudio_Shell_DialogPage_PageApplyEventArgsAccessor e) {
        object e_val_target = null;
        if ((e != null)) {
            e_val_target = e.Target;
        }
        object[] args = new object[] {
                e_val_target};
        Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType target = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs");
        m_privateObject.Invoke("OnApply", new System.Type[] {
                    target.ReferencedType}, args);
    }
}
[System.Diagnostics.DebuggerStepThrough()]
[System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")]
internal class Microsoft_VisualStudio_Shell_DialogPage_PageApplyEventArgsAccessor : BaseAccessor {
    
    protected static Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType m_privateType = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs");
    
    internal Microsoft_VisualStudio_Shell_DialogPage_PageApplyEventArgsAccessor(object target) : 
            base(target, m_privateType) {
    }
    
    internal global::Microsoft.VisualStudio.Shell.DialogPage.ApplyKind ApplyBehavior {
        get {
            global::Microsoft.VisualStudio.Shell.DialogPage.ApplyKind ret = ((global::Microsoft.VisualStudio.Shell.DialogPage.ApplyKind)(m_privateObject.GetProperty("ApplyBehavior")));
            return ret;
        }
        set {
            m_privateObject.SetProperty("ApplyBehavior", value);
        }
    }
    
    internal static global::System.EventArgs CreatePrivate() {
        object[] args = new object[0];
        Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject priv_obj = new Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs", new System.Type[0], args);
        return ((global::System.EventArgs)(priv_obj.Target));
    }
}
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Options Page/VB/TDD/VSCodeGenAccessors.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************


Imports Microsoft.VisualBasic
Imports System
Imports Microsoft.VisualStudio.TestTools.UnitTesting

Namespace Microsoft.Samples.VisualStudio.IDE.OptionsPage.UnitTests
<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class BaseAccessor

	Protected m_privateObject As Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject

	Protected Sub New(ByVal target As Object, ByVal type As Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType)
		m_privateObject = New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject(target, type)
	End Sub

	Protected Sub New(ByVal type As Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType)
		Me.New(Nothing, type)
	End Sub

	Friend Overridable ReadOnly Property Target() As Object
		Get
			Return m_privateObject.Target
		End Get
	End Property

	Public Overrides Function ToString() As String
		Return Me.Target.ToString()
	End Function

	Public Overrides Overloads Function Equals(ByVal obj As Object) As Boolean
		If GetType(BaseAccessor).IsInstanceOfType(obj) Then
			obj = (CType(obj, BaseAccessor)).Target
		End If
		Return Me.Target.Equals(obj)
	End Function

	Public Overrides Function GetHashCode() As Integer
		Return Me.Target.GetHashCode()
	End Function
End Class


<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPagePackageAccessor
	Inherits BaseAccessor

        Protected Shared m_privateType As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(GetType(Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPagePackageVB))

        Friend Sub New(ByVal target As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPagePackageVB)
            MyBase.New(target, m_privateType)
        End Sub

	Friend Sub Initialize()
            Dim args As Object() = New Object() {}
		m_privateObject.Invoke("Initialize", New System.Type(){}, args)
	End Sub
End Class
<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageCustomAccessor
	Inherits BaseAccessor

        Protected Shared m_privateType As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(GetType(Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom))

	Friend Sub New(ByVal target As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom)
		MyBase.New(target, m_privateType)
	End Sub

	Friend Property selectedImagePath() As String
		Get
			Dim ret As String = (CStr(m_privateObject.GetField("selectedImagePath")))
			Return ret
		End Get
		Set(ByVal value As String)
			m_privateObject.SetField("selectedImagePath", value)
		End Set
	End Property

	Friend ReadOnly Property Window() As Global.System.Windows.Forms.IWin32Window
		Get
			Dim ret As Global.System.Windows.Forms.IWin32Window = (CType(m_privateObject.GetProperty("Window"), Global.System.Windows.Forms.IWin32Window))
			Return ret
		End Get
	End Property
End Class
<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsCompositeControlAccessor
	Inherits BaseAccessor

        Protected Shared m_privateType As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(GetType(Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsCompositeControl))

	Friend Sub New(ByVal target As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsCompositeControl)
		MyBase.New(target, m_privateType)
	End Sub

	Friend Property pictureBox() As Global.System.Windows.Forms.PictureBox
		Get
			Dim ret As Global.System.Windows.Forms.PictureBox = (CType(m_privateObject.GetField("pictureBox"), Global.System.Windows.Forms.PictureBox))
			Return ret
		End Get
		Set(ByVal value As Global.System.Windows.Forms.PictureBox)
			m_privateObject.SetField("pictureBox", value)
		End Set
	End Property

	Friend Property openImageFileDialog() As Global.System.Windows.Forms.OpenFileDialog
		Get
			Dim ret As Global.System.Windows.Forms.OpenFileDialog = (CType(m_privateObject.GetField("openImageFileDialog"), Global.System.Windows.Forms.OpenFileDialog))
			Return ret
		End Get
		Set(ByVal value As Global.System.Windows.Forms.OpenFileDialog)
			m_privateObject.SetField("openImageFileDialog", value)
		End Set
	End Property

	Friend Property buttonChooseImage() As Global.System.Windows.Forms.Button
		Get
			Dim ret As Global.System.Windows.Forms.Button = (CType(m_privateObject.GetField("buttonChooseImage"), Global.System.Windows.Forms.Button))
			Return ret
		End Get
		Set(ByVal value As Global.System.Windows.Forms.Button)
			m_privateObject.SetField("buttonChooseImage", value)
		End Set
	End Property

	Friend Property buttonClearImage() As Global.System.Windows.Forms.Button
		Get
			Dim ret As Global.System.Windows.Forms.Button = (CType(m_privateObject.GetField("buttonClearImage"), Global.System.Windows.Forms.Button))
			Return ret
		End Get
		Set(ByVal value As Global.System.Windows.Forms.Button)
			m_privateObject.SetField("buttonClearImage", value)
		End Set
	End Property

	Friend Property customOptionsPage() As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom
		Get
			Dim ret As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom = (CType(m_privateObject.GetField("customOptionsPage"), Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom))
			Return ret
		End Get
		Set(ByVal value As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageCustom)
			m_privateObject.SetField("customOptionsPage", value)
		End Set
	End Property

	Friend Property components() As Global.System.ComponentModel.Container
		Get
			Dim ret As Global.System.ComponentModel.Container = (CType(m_privateObject.GetField("components"), Global.System.ComponentModel.Container))
			Return ret
		End Get
		Set(ByVal value As Global.System.ComponentModel.Container)
			m_privateObject.SetField("components", value)
		End Set
	End Property

	Friend Sub Dispose(ByVal disposing As Boolean)
            Dim args As Object() = New Object() {disposing}
		m_privateObject.Invoke("Dispose", New System.Type() { GetType(Boolean)}, args)
	End Sub

	Friend Sub InitializeComponent()
            Dim args As Object() = New Object() {}
		m_privateObject.Invoke("InitializeComponent", New System.Type(){}, args)
	End Sub

	Friend Sub OnChooseImage(ByVal sender As Object, ByVal e As Global.System.EventArgs)
            Dim args As Object() = New Object() {sender, e}
		m_privateObject.Invoke("OnChooseImage", New System.Type() { GetType(Object), GetType(Global.System.EventArgs)}, args)
	End Sub

	Friend Sub OnClearImage(ByVal sender As Object, ByVal e As Global.System.EventArgs)
		Dim args As Object() = New Object() { sender, e}
		m_privateObject.Invoke("OnClearImage", New System.Type() { GetType(Object), GetType(Global.System.EventArgs)}, args)
	End Sub

	Friend Sub RefreshImage()
		Dim args As Object() = New Object(){}
		m_privateObject.Invoke("RefreshImage", New System.Type(){}, args)
	End Sub
End Class
<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class Microsoft_Samples_VisualStudio_IDE_OptionsPage_OptionsPageGeneralAccessor
	Inherits BaseAccessor

        Protected Shared m_privateType As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType(GetType(Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageGeneral))

	Friend Sub New(ByVal target As Global.Microsoft.Samples.VisualStudio.IDE.OptionsPage.OptionsPageGeneral)
		MyBase.New(target, m_privateType)
	End Sub

	Friend Property optionCustomInteger() As Integer
		Get
			Dim ret As Integer = (CInt(Fix(m_privateObject.GetField("optionCustomInteger"))))
			Return ret
		End Get
		Set(ByVal value As Integer)
			m_privateObject.SetField("optionCustomInteger", value)
		End Set
	End Property

	Friend Property optionCustomSize() As Global.System.Drawing.Size
		Get
			Dim ret As Global.System.Drawing.Size = (CType(m_privateObject.GetField("optionCustomSize"), Global.System.Drawing.Size))
			Return ret
		End Get
		Set(ByVal value As Global.System.Drawing.Size)
			m_privateObject.SetField("optionCustomSize", value)
		End Set
	End Property

	Friend Property optionCustomString() As String
		Get
			Dim ret As String = (CStr(m_privateObject.GetField("optionCustomString")))
			Return ret
		End Get
		Set(ByVal value As String)
			m_privateObject.SetField("optionCustomString", value)
		End Set
	End Property

	Friend Sub OnActivate(ByVal e As Global.System.ComponentModel.CancelEventArgs)
		Dim args As Object() = New Object() { e}
		m_privateObject.Invoke("OnActivate", New System.Type() { GetType(Global.System.ComponentModel.CancelEventArgs)}, args)
	End Sub

	Friend Sub OnClosed(ByVal e As Global.System.EventArgs)
		Dim args As Object() = New Object() { e}
		m_privateObject.Invoke("OnClosed", New System.Type() { GetType(Global.System.EventArgs)}, args)
	End Sub

	Friend Sub OnDeactivate(ByVal e As Global.System.ComponentModel.CancelEventArgs)
		Dim args As Object() = New Object() { e}
		m_privateObject.Invoke("OnDeactivate", New System.Type() { GetType(Global.System.ComponentModel.CancelEventArgs)}, args)
	End Sub

	Friend Sub OnApply(ByVal e As Microsoft_VisualStudio_Shell_DialogPage_PageApplyEventArgsAccessor)
		Dim e_val_target As Object = Nothing
            If (e IsNot Nothing) Then
                e_val_target = e.Target
            End If
		Dim args As Object() = New Object() { e_val_target}
            Dim target As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs")
		m_privateObject.Invoke("OnApply", New System.Type() { target.ReferencedType}, args)
	End Sub
End Class
<System.Diagnostics.DebuggerStepThrough(), System.CodeDom.Compiler.GeneratedCodeAttribute("Microsoft.VisualStudio.TestTools.UnitTestGeneration", "1.0.0.0")> _
Friend Class Microsoft_VisualStudio_Shell_DialogPage_PageApplyEventArgsAccessor
	Inherits BaseAccessor

        Protected Shared m_privateType As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateType("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs")

	Friend Sub New(ByVal target As Object)
		MyBase.New(target, m_privateType)
	End Sub

	Friend Property ApplyBehavior() As Global.Microsoft.VisualStudio.Shell.DialogPage.ApplyKind
		Get
			Dim ret As Global.Microsoft.VisualStudio.Shell.DialogPage.ApplyKind = (CType(m_privateObject.GetProperty("ApplyBehavior"), Global.Microsoft.VisualStudio.Shell.DialogPage.ApplyKind))
			Return ret
		End Get
		Set(ByVal value As Global.Microsoft.VisualStudio.Shell.DialogPage.ApplyKind)
			m_privateObject.SetProperty("ApplyBehavior", value)
		End Set
	End Property

	Friend Shared Function CreatePrivate() As Global.System.EventArgs
		Dim args As Object() = New Object(){}
            Dim priv_obj As New Microsoft.VisualStudio.TestTools.UnitTesting.PrivateObject("Microsoft.VisualStudio.Shell", "Microsoft.VisualStudio.Shell.DialogPage+PageApplyEventArgs", New System.Type() {}, args)
		Return (CType(priv_obj.Target, Global.System.EventArgs))
	End Function
End Class
End Namespace
